core/app/components/FormBuilder/ElementPanel/InputComponentConfigurationForm.tsx

```diff
@@ -84,24 +153,39 @@ const ComponentSelect = ({
 	element: InputElement;
 }) => {
 	return (
-		<RadioGroup
-			className="grid grid-cols-2 gap-3"
-			defaultValue={value}
-			onValueChange={onChange}
-		>
+		<div className="grid grid-cols-2 gap-3">
 			{components.map((c) => {
 				const { name, demoComponent: Component } = componentInfo[c];
 				return (
-					<RadioGroupCard key={c} value={c} className="flex h-[124px] w-full flex-col">
-						<div className="flex h-[88px] w-full items-center justify-center p-3">
-							{Component && <Component element={element} />}
+					<div key={c}>
+						{/* We use regular input instead of a RadioGroup here because the RadioGroup
+						renders buttons, and we cannot render buttons inside of buttons. Some of the demo components
+						need to render buttons. */}
```

Comment:
good catch! this seems to work about as well as the cards did although there's a small regression where clicking inside the demo component's inputs doesn't bubble up to select a different radio input.

i think it might be ok to merge as is but if so we should add a ticket to improve these a bit by making it harder to interact with the demo component inputs through some or all of:
- removing them from the tab sequencing
- maybe disabling the inputs when they're not in a currently selected component?
- bubbling click events up

Reply:
I learned about [inert](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inert) today which seems like a pretty good use case for this—we don't _really_ want those demo components interactive, but they can be if they're currently selected, so that the user can get a feel for them I guess?

here's the behavior now. so they are only part of the tab order if the item is selected. maybe they shouldn't be part of tab order at all? 🤔 but at least clicking on a radio button selects the component selector, if it wasn't selected already, and only if you click it again can you really click the radio

https://github.com/user-attachments/assets/b95da185-ec94-4660-af6f-04cf0dad395a

Reply:
very nice, I'm also just learning about inert for the first time! seems like that is exactly what we need!

Reply:
taking them out of the tab order entirely would be great but i think this is fine for now!